stdlib/public/Cxx/CxxConvertibleToCollection.swift

```diff
@@ -21,36 +21,36 @@ public protocol CxxConvertibleToCollection {
   mutating func __endUnsafe() -> RawIterator
 }
 
-@inlinable @inline(__always)
-internal func forEachElement<C: CxxConvertibleToCollection>(
-  of c: C,
-  body: (C.RawIterator.Pointee) -> Void
-) {
-  var mutableC = c
-  withExtendedLifetime(mutableC) {
-    var rawIterator = mutableC.__beginUnsafe()
-    let endIterator = mutableC.__endUnsafe()
-    while rawIterator != endIterator {
-      body(rawIterator.pointee)
-      rawIterator = rawIterator.successor()
+extension CxxConvertibleToCollection {
+  @inlinable
+  internal func forEach(_ body: (RawIterator.Pointee) -> Void) {
+    var mutableSelf = self
+    withExtendedLifetime(mutableSelf) {
+      var rawIterator = mutableSelf.__beginUnsafe()
+      let endIterator = mutableSelf.__endUnsafe()
+      while rawIterator != endIterator {
+        body(rawIterator.pointee)
+        rawIterator = rawIterator.successor()
+      }
     }
   }
 }
 
-extension Array {
-  /// Creates an array containing the elements of a C++ container.
+extension RangeReplaceableCollection {
+  /// Creates a collection containing the elements of a C++ container.
   ///
-  /// This initializes the array by copying every element of the C++ container.
+  /// This initializes the collection by copying every element of the C++
+  /// container.
   ///
   /// - Complexity: O(*n*), where *n* is the number of elements in the C++
   ///   container when each element is copied in O(1). Note that this might not
   ///   be true for certain C++ types, e.g. those with a custom copy
   ///   constructor that performs additional logic.
-  public init<C: CxxConvertibleToCollection>(_ c: C)
+  public init<C: CxxConvertibleToCollection>(contentsOf c: C)
```

Comment:
Hm, this new label now looks questionable -- I hope we didn't recommend it as a style fix. 🙈 

The naming convention is that these copying/conversion initializers do not use the `contentsOf:` label, even though we do use them for things like `append(contentsOf:)` and `insert(contentsOf:at:)`. (Presumably to help disambiguate them in cases where the element type is a collection of itself.) E.g., `RangeReplaceableCollection` has this unlabeled initializer requirement:

```swift
init(_ elements: some Sequence<Element>)
```

I remember discussing potential name resolution ambiguities when a type implements `CxxConvertibleToCollection` _in addition to_ something like `Sequence`:

```swift
// If `bar` conforms to both, is this calling the `Sequence`
// initializer or the `CxxConvertibleToCollection` one?
let foo = Array(bar)
```

Adding a distinguishing label may would indeed work around that, but if that's the reason for adding it, it would probably be a better idea to use something concrete like `init(cxx:)`, rather than a generic one like this -- to me, this reads too much like we are inconsistent about API naming.

Reply:
Thanks! I removed the label. I'll tackle the name resolution ambiguity in the next patch where I will restructure the protocol inheritance layout in the overlay.